tasks/list-remaining-migrated-removals.ts

```diff
@@ -0,0 +1,207 @@
+  dryRun?: boolean;
+}
+
+type ParsedListMigratedRemovalsTaskOptions = RequiredKeys<
+  ListMigratedRemovalsTaskOptions,
+  'file'
+>;
+
+export const GET_LIST_MIGRATED_REMOVALS_TASK = () =>
+  ({
+    name: 'list-remaining-migrated-removals',
+    description:
+      'Utility to list migrated removals for sale that were not included in migrated certificates',
+    run: async (
+      options: ListMigratedRemovalsTaskOptions,
+      _: CustomHardHatRuntimeEnvironment
+    ): Promise<void> => {
+      const {
+        file,
+        outputFile = 'listed-migrated-removals.json',
+        dryRun,
+      } = options as ParsedListMigratedRemovalsTaskOptions;
+      const jsonData = JSON.parse(readFileSync(file, 'utf8'));
+      // hre.log({ jsonData });
+
+      const [signer] = await hre.getSigners();
+      const signerAddress = await signer.getAddress();
+      const { getRemoval } = await import('@/utils/contracts');
+      const removalContract = await getRemoval({
+        hre,
+        signer,
+      });
+      hre.log(`Removal contract address: ${removalContract.address}`);
+      hre.log(`Signer address: ${signerAddress}`);
+      // const fireblocksSigner = removalContract.signer as FireblocksSigner;
+
+      if (!dryRun) {
+        hre.log(
+          chalk.bold.white(
+            `✨ Listing unsold removals for ${jsonData.length} projects...`
+          )
+        );
+      } else {
+        hre.log(
+          chalk.bold.white(
+            `DRY RUN 🌵 Listing unsold removals for ${jsonData.length} projects...`
+          )
+        );
+      }
+
+      const allMigratedRemovalIds = jsonData.flatMap(
+        (project) => project.tokenIds
+      );
+      if (allMigratedRemovalIds.includes(undefined)) {
+        hre.log(
+          chalk.bold.red(
+            `❌ Some migrated projects have undefined token ids. Please check the input file for transaction errors during minting. Exiting...`
+          )
+        );
+        return;
+      }
+
+      hre.log(chalk.white(`👀 Querying unsold removal balances...`));
+
+      const remainingBalanceData = await Promise.all(
+        allMigratedRemovalIds.map(async (tokenId) => {
+          const balance = await removalContract.balanceOf(
+            signerAddress,
+            tokenId
+          );
+          return { tokenId, balance };
+        })
+      );
+      // filter out token ids that have a balance of 0
+      const listableData = remainingBalanceData.filter((data) =>
+        data.balance.gt(0)
+      );
+
+      const listableTokenIds = listableData.map((data) => data.tokenId);
+      const listableBalances = listableData.map((data) => data.balance);
+      // sum the listable balances and convert to ether
+      const totalListableBalance = listableBalances.reduce(
+        (accumulator, balance) => accumulator.add(balance),
+        ethers.BigNumber.from(0)
+      );
+      const totalListableBalanceInEther =
+        ethers.utils.formatEther(totalListableBalance);
```

Comment:
```suggestion
        formatTokenAmount(totalListableBalance);
```

Reply:
this one goes from ether to wei, but we want wei to ether